core/views.py

```diff
@@ -1,22 +1,113 @@
+    post:
+    Creates a new team in a hackathon and return the team_id
+    """
+    permission_classes = [permissions.IsAuthenticated, AllowCompleteProfile]
+
+    def get_serializer_class(self):
+        if self.request.method == 'GET':
+            return TeamSerializer
+        else:
+            return TeamCreateSerializer
+    def get_serializer_context(self):
+        return {
+            'request': self.request,
+            'kwargs': self.kwargs
+        }
+
+    def get_queryset(self, **kwargs):
+        if getattr(self, 'swagger_fake_view', False):
+            return None
+        try:
+            hackathon = Hackathon.objects.get(id=self.kwargs['pk'])
+        except Hackathon.DoesNotExist:
+            raise exceptions.NotFound("Hackathon does not exist!")
+        queryset = Team.objects.filter(hackathon=self.kwargs['pk'])
+        return queryset
+
+    def post(self, request, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        team = serializer.save()
+        serializer = TeamSerializer(team)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+class JoinTeamView(generics.GenericAPIView):
+    """
+    patch:
+    Join a team with team_id and hackathon_id
+    """
+    serializer_class = JoinTeamSerializer
+
+    def get_queryset(self):
+        if getattr(self, 'swagger_fake_view', False):
+            return None
+        pass
+
+    def get_serializer_context(self):
+        return {
+            'request': self.request,
+            'kwargs': self.kwargs
+        }
+
+    def patch(self, request, **kwargs):
+        serializer = self.get_serializer()
+        serializer.join_team()
+        return Response("Successfully jonied team!",status=status.HTTP_200_OK)
+
+query_param = openapi.Parameter(
+    'query', openapi.IN_QUERY, description="Query parameter - Returns all hackthons if not specified.",
+    type=openapi.TYPE_STRING, enum=['completed', 'upcoming', 'ongoing'])
+@method_decorator(name="get", decorator=swagger_auto_schema(manual_parameters=[query_param]))
+class HackathonListCreateView(generics.ListCreateAPIView):
+    """
+    get:
+    Returns list of Hackathons according to query parameter.
+
+    post:
+    Creates a new hackathon. Only admin can create a hackathon
+    """
     serializer_class = HackathonSerializer
-    permission_classes = [permissions.IsAdminUser]
+
+    def get_permissions(self):
+        if self.request.method == "GET":
+            return [permissions.AllowAny()]
+        else:
+            return [permissions.IsAdminUser()]
+
+    def get_queryset(self):
+        queryset = Hackathon.objects.all()
+        query = self.request.query_params.get('query', None)
+        current_date = timezone.now()
+        if query is not None:
+            if(query == 'ongoing'):
+                queryset = Hackathon.objects.filter(start__lt=current_date, end__gt=current_date)
```

Comment:
```suggestion
                queryset = Hackathon.objects.filter(start__lte=current_date, end__gte=current_date)
```
A single-day hackathon might get excluded in 'ongoing'

Reply:
Correct, I missed this. Will change